src/test/scala/kinesis/mock/api/PutRecordsTests.scala

```diff
@@ -38,7 +50,10 @@ class PutRecordsTests
         res.isRight && s.streams.get(streamName).exists { stream =>
           stream.shards.values.toVector.flatten.count { rec =>
             req.records.map(_.data).exists(_.sameElements(rec.data))
-          } == initReq.records.length
+          } == initReq.records.length && res.exists { r =>
+            r.records.map(_.shardId) == predictedShards
```

Comment:
Added some tests here for ensuring that the shard-id in the returned records are the same as the predicted shard ids.